File: src/fortran/ElsetRec.f90
MODULE ElsetRecMod

TYPE ElsetRec 
    INTEGER*4 whichconst
    INTEGER*4 satnum
    INTEGER*4 epochyr
    INTEGER*4 epochtynumrev
    INTEGER*4 error
    CHARACTER operationmode
    CHARACTER init
    CHARACTER method    
    REAL*8 a
    REAL*8 altp
    REAL*8 alta
    REAL*8 epochdays
    REAL*8 jdsatepoch
    REAL*8 jdsatepochF
    REAL*8 nddot
    REAL*8 ndot
    REAL*8 bstar
    REAL*8 rcse
    REAL*8 inclo
    REAL*8 nodeo
    REAL*8 ecco
    REAL*8 argpo
    REAL*8 mo
    REAL*8 no_kozai
    
!   // sgp4fix add new variables from tle
    CHARACTER classification
    CHARACTER*12 intldesg
    INTEGER*4 ephtype
    INTEGER*8 elnum
    INTEGER*8 revnum
    
!   // sgp4fix add unkozai'd variable
    REAL*8 no_unkozai
    
!   // sgp4fix add singly averaged variables
    REAL*8 am
    REAL*8 em
    REAL*8 im
    REAL*8 OOm
    REAL*8 om
    REAL*8 mm
    REAL*8 nm
    REAL*8 t
    
!   // sgp4fix add constant parameters to eliminate mutliple calls during execution
    REAL*8 tumin
    REAL*8 mu
    REAL*8 radiusearthkm
    REAL*8 xke 
    REAL*8 j2
    REAL*8 j3
    REAL*8 j4
    REAL*8 j3oj2
    
!   //       Additional elements to capture relevant TLE and object information:       
    INTEGER*8 dia_mm 
    REAL*8 period_sec 
    CHARACTER active 
    CHARACTER not_orbital 
    REAL*8 rcs_m2 

!   // temporary variables because the original authors call the same method with different variables
    REAL*8 ep
    REAL*8 inclp
    REAL*8 nodep
    REAL*8 argpp
    REAL*8 mp


    INTEGER*4 isimp
    REAL*8 aycof
    REAL*8 con41
    REAL*8 cc1
    REAL*8 cc4
    REAL*8 cc5
    REAL*8 d2
    REAL*8 d3
    REAL*8 d4
    REAL*8 delmo
    REAL*8 eta
    REAL*8 argpdot
    REAL*8 omgcof
    REAL*8 sinmao
    REAL*8 t2cof
    REAL*8 t3cof
    REAL*8 t4cof
    REAL*8 t5cof
    REAL*8 x1mth2
    REAL*8 x7thm1
    REAL*8 mdot
    REAL*8 nodedot
    REAL*8 xlcof
    REAL*8 xmcof
    REAL*8 nodecf

!   // deep space
    INTEGER*4 irez
    REAL*8 d2201
    REAL*8 d2211
    REAL*8 d3210
    REAL*8 d3222
    REAL*8 d4410
    REAL*8 d4422
    REAL*8 d5220
    REAL*8 d5232
    REAL*8 d5421
    REAL*8 d5433
    REAL*8 dedt
    REAL*8 del1
    REAL*8 del2
    REAL*8 del3
    REAL*8 didt
    REAL*8 dmdt
    REAL*8 dnodt
    REAL*8 domdt
    REAL*8 e3
    REAL*8 ee2
    REAL*8 peo
    REAL*8 pgho
    REAL*8 pho
    REAL*8 pinco
    REAL*8 plo
    REAL*8 se2
    REAL*8 se3
    REAL*8 sgh2
    REAL*8 sgh3
    REAL*8 sgh4
    REAL*8 sh2
    REAL*8 sh3
    REAL*8 si2
    REAL*8 si3
    REAL*8 sl2
    REAL*8 sl3
    REAL*8 sl4
    REAL*8 gsto
    REAL*8 xfact
    REAL*8 xgh2
    REAL*8 xgh3
    REAL*8 xgh4
    REAL*8 xh2
    REAL*8 xh3
    REAL*8 xi2
    REAL*8 xi3
    REAL*8 xl2
    REAL*8 xl3
    REAL*8 xl4
    REAL*8 xlamo
    REAL*8 zmol
    REAL*8 zmos
    REAL*8 atime
    REAL*8 xli
    REAL*8 xni
    REAL*8 snodm
    REAL*8 cnodm
    REAL*8 sinim
    REAL*8 cosim
    REAL*8 sinomm
    REAL*8 cosomm
    REAL*8 day
    REAL*8 emsq
    REAL*8 gam
    REAL*8 rtemsq 
    REAL*8 s1
    REAL*8 s2
    REAL*8 s3
    REAL*8 s4
    REAL*8 s5
    REAL*8 s6
    REAL*8 s7 
    REAL*8 ss1
    REAL*8 ss2
    REAL*8 ss3
    REAL*8 ss4
    REAL*8 ss5
    REAL*8 ss6
    REAL*8 ss7
    REAL*8 sz1
    REAL*8 sz2
    REAL*8 sz3
    REAL*8 sz11
    REAL*8 sz12
    REAL*8 sz13
    REAL*8 sz21
    REAL*8 sz22
    REAL*8 sz23
    REAL*8 sz31
    REAL*8 sz32
    REAL*8 sz33
    REAL*8 z1
    REAL*8 z2
    REAL*8 z3
    REAL*8 z11
    REAL*8 z12
    REAL*8 z13
    REAL*8 z21
    REAL*8 z22
    REAL*8 z23
    REAL*8 z31
    REAL*8 z32
    REAL*8 z33
    REAL*8 argpm
    REAL*8 inclm
    REAL*8 nodem
    REAL*8 dndt
    REAL*8 eccsq
        
!   // for initl
    REAL*8 ainv
    REAL*8 ao
    REAL*8 con42
    REAL*8 cosio
    REAL*8 cosio2
    REAL*8 omeosq
    REAL*8 posq
    REAL*8 rp
    REAL*8 rteosq
    REAL*8 sinio
  END TYPE ElsetRec

END MODULE ElsetRecMod

File: src/python/ElsetRec.py
#
# This class implements the elsetrec data type from Vallado's SGP4 code.
# 
# From SGP4.h
# #define SGP4Version  "SGP4 Version 2016-03-09"
# 
# @author aholinch
#
#
class ElsetRec:
  def __init__(self):

    self.whichconst = 2    #SGP4.wgs72
    self.satid = None
    self.epochyr = 0
    self.epochtynumrev = 0
    self.error = 0
    self.operationmode = 0
    self.init = 0
    self.method = 0    
    self.a = 0.0
    self.altp = 0.0
    self.alta = 0.0
    self.epochdays = 0.0
    self.jdsatepoch = 0.0
    self.jdsatepochF = 0.0
    self.nddot = 0.0
    self.ndot = 0.0
    self.bstar = 0.0
    self.rcse = 0.0
    self.inclo = 0.0
    self.nodeo = 0.0
    self.ecco = 0.0
    self.argpo = 0.0
    self.mo = 0.0
    self.no_kozai = 0.0
    
    # sgp4fix add new variables from tle
    self.classification = 'U'
    self.intldesg = ""
    self.ephtype = 0
    self.elnum = 0
    self.revnum = 0
    
    # sgp4fix add unkozai'd variable
    self.gno_unkozai = 0.0
    # sgp4fix add singly averaged variables
    self.am = 0.0
    self.em = 0.0
    self.im = 0.0
    self.Om = 0.0
    self.om = 0.0
    self.mm = 0.0
    self.nm = 0.0
    self.t = 0.0
    
    # sgp4fix add constant parameters to eliminate mutliple calls during execution
    self.tumin = 0.0
    self.mu = 0.0
    self.radiusearthkm = 0.0
    self.xke = 0.0 
    self.j2 = 0.0
    self.j3 = 0.0
    self.j4 = 0.0
    self.j3oj2 = 0.0
    
    #       Additional elements to capture relevant TLE and object information:       
    self.dia_mm = 0 # RSO dia in mm
    self.period_sec = 0.0 # Period in seconds
    self.active = 0 # "Active S/C" flag (0=n, 1=y) 
    self.not_orbital = 0 # "Orbiting S/C" flag (0=n, 1=y)  
    self.rcs_m2 = 0.0 # "RCS (m^2)" storage  

    # temporary variables because the original authors call the same method with different variables
    self.ep = 0.0
    self.inclp = 0.0
    self.nodep = 0.0
    self.argpp = 0.0
    self.mp = 0.0


    self.isimp = 0
    self.aycof = 0.0
    self.con41 = 0.0
    self.cc1 = 0.0
    self.cc4 = 0.0
    self.cc5 = 0.0
    self.d2 = 0.0
    self.d3 = 0.0
    self.d4 = 0.0
    self.delmo = 0.0
    self.eta = 0.0
    self.argpdot = 0.0
    self.omgcof = 0.0
    self.sinmao = 0.0
    self.t2cof = 0.0
    self.t3cof = 0.0
    self.t4cof = 0.0
    self.t5cof = 0.0
    self.x1mth2 = 0.0
    self.x7thm1 = 0.0
    self.mdot = 0.0
    self.nodedot = 0.0
    self.xlcof = 0.0
    self.xmcof = 0.0
    self.nodecf = 0.0

    # deep space
    self.irez = 0
    self.d2201 = 0.0
    self.d2211 = 0.0
    self.d3210 = 0.0
    self.d3222 = 0.0
    self.d4410 = 0.0
    self.d4422 = 0.0
    self.d5220 = 0.0
    self.d5232 = 0.0
    self.d5421 = 0.0
    self.d5433 = 0.0
    self.dedt = 0.0
    self.del1 = 0.0
    self.del2 = 0.0
    self.del3 = 0.0
    self.didt = 0.0
    self.dmdt = 0.0
    self.dnodt = 0.0
    self.domdt = 0.0
    self.e3 = 0.0
    self.ee2 = 0.0
    self.peo = 0.0
    self.pgho = 0.0
    self.pho = 0.0
    self.pinco = 0.0
    self.plo = 0.0
    self.se2 = 0.0
    self.se3 = 0.0
    self.sgh2 = 0.0
    self.sgh3 = 0.0
    self.sgh4 = 0.0
    self.sh2 = 0.0
    self.sh3 = 0.0
    self.si2 = 0.0
    self.si3 = 0.0
    self.sl2 = 0.0
    self.sl3 = 0.0
    self.sl4 = 0.0
    self.gsto = 0.0
    self.xfact = 0.0
    self.xgh2 = 0.0
    self.xgh3 = 0.0
    self.xgh4 = 0.0
    self.xh2 = 0.0
    self.xh3 = 0.0
    self.xi2 = 0.0
    self.xi3 = 0.0
    self.xl2 = 0.0
    self.xl3 = 0.0
    self.xl4 = 0.0
    self.xlamo = 0.0
    self.zmol = 0.0
    self.zmos = 0.0
    self.atime = 0.0
    self.xli = 0.0
    self.xni = 0.0
    self.snodm = 0.0
    self.cnodm = 0.0
    self.sinim = 0.0
    self.cosim = 0.0
    self.sinomm = 0.0
    self.cosomm = 0.0
    self.day = 0.0
    self.emsq = 0.0
    self.gam = 0.0
    self.rtemsq = 0.0 
    self.s1 = 0.0
    self.s2 = 0.0
    self.s3 = 0.0
    self.s4 = 0.0
    self.s5 = 0.0
    self.s6 = 0.0
    self.s7 = 0.0 
    self.ss1 = 0.0
    self.ss2 = 0.0
    self.ss3 = 0.0
    self.ss4 = 0.0
    self.ss5 = 0.0
    self.ss6 = 0.0
    self.ss7 = 0.0
    self.sz1 = 0.0
    self.sz2 = 0.0
    self.sz3 = 0.0
    self.sz11 = 0.0
    self.sz12 = 0.0
    self.sz13 = 0.0
    self.sz21 = 0.0
    self.sz22 = 0.0
    self.sz23 = 0.0
    self.sz31 = 0.0
    self.sz32 = 0.0
    self.sz33 = 0.0
    self.z1 = 0.0
    self.z2 = 0.0
    self.z3 = 0.0
    self.z11 = 0.0
    self.z12 = 0.0
    self.z13 = 0.0
    self.z21 = 0.0
    self.z22 = 0.0
    self.z23 = 0.0
    self.z31 = 0.0
    self.z32 = 0.0
    self.z33 = 0.0
    self.argpm = 0.0
    self.inclm = 0.0
    self.nodem = 0.0
    self.dndt = 0.0
    self.eccsq = 0.0
        
    # for initl
    self.ainv = 0.0
    self.ao = 0.0
    self.con42 = 0.0
    self.cosio = 0.0
    self.cosio2 = 0.0
    self.omeosq = 0.0
    self.posq = 0.0
    self.rp = 0.0
    self.rteosq = 0.0
    self.sinio = 0.0
